### PR TITLE
[Topic Page] Add bar tile component

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: static/js/topic_page/bar_tile.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2021 Google LLC
```
Comment (beets) [inline static/js/topic_page/bar_tile.tsx:2]:
```suggestion
 * Copyright 2022 Google LLC
```
:)

**Review by beets — APPROVED** [review]
thanks!

File: static/js/topic_page/bar_tile.tsx (hunk 1 above)
Comment (shifucun) [reply static/js/topic_page/bar_tile.tsx]: Oh!

File: static/js/topic_page/bar_tile.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  useEffect(() => {
+    fetchData(props, setRawData);
+  }, [props]);
+
+  useEffect(() => {
+    if (rawData) {
+      processData(props, rawData, setChartData);
+    }
+  }, [props, rawData]);
```
Comment (chejennifer) [inline static/js/topic_page/bar_tile.tsx:57]: I think we would only need to re-run processData when rawData changes so I don't think this useEffect needs to also listen for props changes. Same with the following useEffect that does the drawing, I think we only need to re-run draw when the chartData changes

**Review by chejennifer — APPROVED** [review]
nice!

File: static/js/topic_page/bar_tile.tsx (hunk 2 above)
Comment (shifucun) [reply static/js/topic_page/bar_tile.tsx:57]: This is from the eslint-plugin-react-hooks plugin. There are some discussions about in this thread https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528.

I yet to find sometime to fully understand it, but from initial impression, it's better to include any props, state that this hook refers.

Comment (chejennifer) [reply static/js/topic_page/bar_tile.tsx:57]: oh, interesting, I didn't know that!

Comment (shifucun) [reply static/js/topic_page/bar_tile.tsx:57]: If a prop change, it should be a re-render from parent component, then this hook should be triggered too.

**Pull request merged**